File: src/main/java/io/github/icusystem/icu_connect/api_icu/ApiService.java
package io.github.icusystem.icu_connect.api_icu;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

import java.util.ArrayList;
import java.util.HashMap;

public interface ApiService {

    /**
     * @param params hashmap of API authentication parameters
     * @return Token object
     */
    @FormUrlEncoded
    @POST("/Token")
    Call<Token> getToken(@FieldMap HashMap<String, String> params);

    /**
     * @param token the authorization token value
     * @return ICU device StatusResponse object
     */
    @GET("/api/v1_0/status")
    Call<StatusResponse> getStatus(@Header("Authorization") String token);

    /**
     * @param token the authorization token value
     * @param image the EnrollRequest object with image data
     * @return EnrollResponse object
     */
    @POST("/api/v1_0/image")
    Call<ArrayList<EnrollItem>> getImageUID(@Header("Authorization") String token, @Body EnrollRequest image);

    @POST("/api/v1_0/imageupdate")
    Call<Void> setImageUpdate(@Header("Authorization") String token, @Body UpdateFaceData image);

    @POST("/api/v1_0/imagedelete")
    Call<Void> setImageDelete(@Header("Authorization") String token, @Body FaceDelete image);

    @GET("/api/v1_0/settings")
    Call<DeviceSettings> getSettings(@Header("Authorization") String token);

    @GET("/api/v1_0/device")
    Call<DeviceDetail> getDevice(@Header("Authorization") String token);

    @POST("/api/v1_0/settings")
    Call<Void> setSettings(@Header("Authorization") String token, @Body SettingsRequest settings);

    @POST("/api/v1_0/settings")
    Call<Void> setSettings(@Header("Authorization") String token, @Body ModeSet settings);

    @POST("/api/v1_0/session")
    Call<Void> setSession(@Header("Authorization") String token, @Body SetSession session);

    @GET("/api/v1_0/session")
    Call<SessionResponse> getSession(@Header("Authorization") String token);

    @GET("/api/v1_0/session_age_result")
    Call<SessionAgeResult> getSessionAgeResult(@Header("Authorization") String token);

    @GET("/api/v1_0/session_scan_result")
    Call<SessionScanResult> getSessionScanResult(@Header("Authorization") String token);

    @POST("/api/v1_0/streamsettings")
    Call<Void> setStreamSettings(@Header("Authorization") String token, @Body StreamSetFaceBox stream);
}

File: src/main/java/com/itl/icu_connect/api_icu/ApiService.kt
package com.itl.icu_connect.api_icu

import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.*


interface ApiService {


    /**
     * @param params hashmap of API authentication parameters
     * @return Token object
     */
    @FormUrlEncoded
    @POST("/Token")
    fun getToken(@FieldMap params: HashMap<String, String>): Call<Token>

    /**
     * @param token the authorization token value
     * @return ICU device StatusResponse object
     */
    @GET("/api/v1_0/status")
    fun getStatus(@Header("Authorization") token: String): Call<StatusResponse>

    /**
     * @param token the authorization token value
     * @param faceDelete the Array of strings object with ID deletion values
     * @return void
     */
    @POST("/api/v1_0/imagedelete")
    fun getImageUID(@Header("Authorization") token: String?, @Body image: EnrollRequest): Call<EnrollResponse>


    @POST("/api/v1_0/imageupdate")
    fun setImageUpdate(@Header("Authorization") token: String, @Body image: UpdateFaceData): Call<Void>


    @POST("/api/v1_0/imagedelete")
    fun setImageDelete(@Header("Authorization") token: String, @Body image: FaceDelete): Call<Void>


    @GET("/api/v1_0/settings")
    fun getSettings(@Header("Authorization") token: String): Call<DeviceSettings>

    @GET("/api/v1_0/device")
    fun getDevice(@Header("Authorization") token: String): Call<DeviceDetail>

    @POST("/api/v1_0/settings")
    fun setSettings(@Header("Authorization") token: String, @Body settings: SettingsRequest): Call<Void>

    @POST("/api/v1_0/settings")
    fun setSettings(@Header("Authorization") token: String, @Body settings: ModeSet): Call<Void>

    @POST("/api/v1_0/session")
    fun setSession(@Header("Authorization") token: String, @Body session: SetSession): Call<Void>

    @GET("/api/v1_0/session")
    fun getSession(@Header("Authorization") token: String): Call<SessionResponse>

    @GET("/api/v1_0/session_age_result")
    fun getSessionAgeResult(@Header("Authorization") token: String): Call<SessionAgeResult>

    @GET("/api/v1_0/session_scan_result")
    fun getSessionScanResult(@Header("Authorization") token: String): Call<SessionScanResult>

    @POST("/api/v1_0/streamsettings")
    fun setStreamSettings(@Header("Authorization") token: String, @Body stream: StreamSetFaceBox): Call<Void>


}